xpdview/waterfall.py

```diff
@@ -94,96 +85,80 @@ def update(self, key_list=None, int_data_list=None, refresh=False):
         refresh : bool, optional
             option to set refresh or not. default to False.
         """
-        if not int_data_list:
-            print("INFO: no reduced data was fed in, "
-                  "waterfall plot can't be updated")
-            self.halt = True
-            self.no_int_data_plot(self.ax, self.canvas)
-            return
-        # refresh list
-        if refresh:
-            self.key_list = []
-            self.int_data_list = []
         self.key_list.extend(key_list)
         self.int_data_list.extend(int_data_list)
-        if len(self.key_list) > self.size:
-            self.key_list.pop(0)
-            self.int_data_list.pop(0)
-        self._adapt_data_list(self.int_data_list)
         # generate plot
-        self.halt = False
+        self._update_data()
         self._update_plot()  # use current value of x,y offset
 
     def _adapt_data_list(self, int_data_list):
-        """method to return statefull information of 1D data list"""
-        x_array_list = []
-        y_array_list = []
+        """method to return stateful information of 1D data list"""
         # parse
+        x_dist, y_dist = 0, 0
         for x, y in int_data_list:
-            x_array_list.append(x)
-            y_array_list.append(y)
-        self.x_array_list = x_array_list
-        self.y_array_list = y_array_list
-        # stateful information
-        self.y_dist = np.max(y_array_list) - np.min(y_array_list)
-        self.x_dist = np.max(x_array_list) - np.min(x_array_list)
+            self.xdist = max(np.ptp(x), self.xdist)
+            self.ydist = max(np.ptp(y), self.ydist)
+            self.x_array_list.append(x)
+            self.y_array_list.append(y)
+        self.x_dist = x_dist
+        self.y_dist = y_dist
 
-    def on_plot_hover(self, event):
-        """callback to show legend when click on one of curves"""
-        line = event.artist
-        name = line.get_label()
-        line.axes.legend([name], handlelength=0,
-                         handletextpad=0, fancybox=True)
-        line.figure.canvas.draw_idle()
+    def _update_data(self):
+        # draw if fresh axes
+        if not self.ax.get_lines():
```

Comment:
wana use `self.ax.lines``? since it directly returns a python list. No difference at all. (also because we call self.ax.lines in clear method)